### DISCUSSION TITLE
Purpose of `label` field on keys

### DISCUSSION THREAD

**Discussion by MitchellJeppson** [liveshowy/webauthn_components#64] [Q&A]
Hey Owen! Let me know if there is a better way for us to communicate. I'm just stuck on what the purpose of the `label` field is on the user_keys. Is this so we could present the user with a textbox and allow them to name the key something they will recognize? Or is there a way to get the name of the authenticator during attestation? I've played around with the Wax.Metadata functions, but couldn't get what I wanted out of it. Appreciate your input! 

Also posting some other small issues I've found here, mostly for myself but feel free to address them if you have thoughts.
* Need to either bump the required elixir version up or change some functions in the generation process. 
* When I try to sign in from my phone via QR and use 1Password it errors out.

**Comment by type1fool** [comment]
# Labels
Right, if a user has multiple keys, they'll want some kind of descriptor. For example, "iCloud Passkey" or "1Password Passkey", maybe even "Yubikey NFC C". If we can detect the authenticator name and provide it as a default, that would be great. This may be possible to obtain from the JS side since that's where the authenticator device is registered.

The user should be able to call it whatever they like.

# Communication
I think this is a good use of GH Discussions. The questions you're asking will be informative to other devs & users.

# Misc

- A separate PR for version updates would be ideal.
- Can you elaborate on "_change some functions in the generation process_"?
- I'll try to reproduce the QR code issue.

**Reply by MitchellJeppson** [reply]
Cool, I will implement the custom lable stuff and then if we find out we can get a sane default value we can add that in later. 

As for the "change some functions in the generation process": We use `NaiveDateTime.utc_now/1` which isn't wasn't added until elixir 1.4.0, so we either need to refactor where we use those, or upgrade the required elixir version. Like you said though, that can all be in another MR. There could be more functions that are breaking as well, that is just one example. Let me know if I can help with the 1Password QR thing 👍
